cmd/eudico/mirvalidator/run.go

```diff
@@ -175,11 +180,23 @@ var runCmd = &cli.Command{
 			log.Info("Initializing mir validator from checkpoint in height: %d", cctx.Int("init-height"))
 		}
 
+		cfg := mir.NewConfig(
+			validatorID,
+			dbPath,
+			initCh,
+			cctx.String("checkpoints-repo"),
+			segmentLength,
+			cctx.String("ipcagent-url"),
+		)
+
 		var mb membership.Reader
 		switch cctx.String("membership") {
 		case "file":
 			mf := filepath.Join(cctx.String("repo"), cctx.String("membership-file"))
 			mb = membership.NewFileMembership(mf)
+		case "onchain":
```

Comment:
This assumes that as part of the command you have to pass the type of membership and the corresponding config as flags, right?

Reply:
No. If the membership type is on-chain then on-chain membership reader will be created and passed to the Mir function as an argument.